boudicca.base/eventcollector-client/src/main/kotlin/base/boudicca/api/eventcollector/collections/CollectionClasses.kt

```diff
@@ -9,43 +8,73 @@ data class FullCollection(
-    val logLines: MutableList<Pair<LogLevel, ByteArray>>,
+    val logLines: MutableList<String>,
+    var errorCount: Int,
+    var warningCount: Int,
 ) {
     constructor() : this(
         UUID.randomUUID(),
         0,
         0,
         Collections.synchronizedList(mutableListOf()),
-        Collections.synchronizedList(mutableListOf())
+        Collections.synchronizedList(mutableListOf()),
+        0,
+        0
     )
 
     override fun toString(): String {
-        return "FullCollection(\nid=$id, \nstartTime=$startTime, \nendTime=$endTime, \nsingleCollections=$singleCollections, \nlogLines=$logLines)"
+        return "FullCollection(\nid=$id, \nstartTime=$startTime, \nendTime=$endTime, \nsingleCollections=$singleCollections, \nlogLines=$logLines, \nerrorCount=$errorCount, \nwarningCount=$warningCount)"
+    }
+
+    /**
+     * gets the error count for this full collection + each single collection
+     */
+    fun getTotalErrorCount(): Int {
+        return errorCount + singleCollections.sumOf { it.errorCount }
+    }
+
+    /**
+     * gets the warning count for this full collection + each single collection
+     */
+    fun getTotalWarningCount(): Int {
+        return warningCount + singleCollections.sumOf { it.warningCount }
+    }
+
+    /**
+     * gets all log lines for this full collection + each single collection. please note that those loglines are not sorted by time but by single-/fullcollection
+     */
+    fun getAllLogLines(): List<String> {
+        return logLines + singleCollections.flatMap { it.logLines }
     }
+
 }
 
 data class SingleCollection(
     val id: UUID,
+    val collectorName: String,
     var startTime: Long,
     var endTime: Long,
-    var collector: EventCollector?,
-    var totalEventsCollected: Int?,
+    var totalEventsCollected: Int,
     val httpCalls: MutableList<HttpCall>,
-    val logLines: MutableList<Pair<LogLevel, ByteArray>>,
+    val logLines: MutableList<String>,
+    var errorCount: Int,
+    var warningCount: Int,
 ) {
-    constructor() : this(UUID.randomUUID(), 0, 0, null, null, mutableListOf(), mutableListOf())
+    constructor(collectorName: String) : this(
+        UUID.randomUUID(),
+        collectorName,
+        0,
+        0,
+        0,
+        mutableListOf(),
+        mutableListOf(),
+        0,
+        0
+    )
 
     override fun toString(): String {
-        return "SingleCollection(\nid=$id, \nstartTime=$startTime, \nendTime=$endTime, \ncollector=$collector, \nhttpCalls=$httpCalls, \nlogLines=${
-            logLines.map {
-                Pair(
-                    it.first,
-                    String(it.second)
-                )
-            }
-        })"
+        return "SingleCollection(\nid=$id, \ncollectorName='$collectorName', \nstartTime=$startTime, \nendTime=$endTime, \ntotalEventsCollected=$totalEventsCollected, \nhttpCalls=$httpCalls, \nlogLines=$logLines, \nerrorCount=$errorCount, \nwarningCount=$warningCount)"
```

Comment:
jfyi Kotlin can do multiline strings which would make something like this more readable 
https://kotlinlang.org/docs/strings.html#multiline-strings

Reply:
hm... probably, but this is was my intellij generated, so I didn't want to change it :D